### PR TITLE
makemake (new formula)

### PR DESCRIPTION
**Pull request by Julow**
Sorry for the duplicate request

### CODE REVIEW

File: Library/Formula/makemake.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# makemake formula
+
+class Makemake < Formula
+
```
Comment (bfontaine) [inline Library/Formula/makemake.rb]: Please remove this empty line

File: Library/Formula/makemake.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# makemake formula
+
```
Comment (bfontaine) [inline Library/Formula/makemake.rb]: Please remove this empty line as well as the top comment.

File: Library/Formula/makemake.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  url "https://github.com/Julow/makemake.git", :revision => "a11bba0903aad98765b7fdac0a67ed09903ba669"
+  head "https://github.com/Julow/makemake.git"
+  # :using => :git is "Redundant"
+
+  sha256 "47b79f953e39db26d4cdd612572c52e9a06b7899025ef53ed874baa4a7a03b23"
+  version "1.0"
+
+  def install
+
```
Comment (bfontaine) [inline Library/Formula/makemake.rb]: Also this one.

File: Library/Formula/makemake.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    mv "makemake.py", "makemake"
+    bin.install "makemake"
+
```
Comment (bfontaine) [inline Library/Formula/makemake.rb]: And this one.

File: Library/Formula/makemake.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  end
+
+  test do
+
```
Comment (bfontaine) [inline Library/Formula/makemake.rb]: And this one.

File: Library/Formula/makemake.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    system "echo", "Makemake is an interative program"
+    system "makemake", "-h"
+
```
Comment (bfontaine) [inline Library/Formula/makemake.rb]: And this one.

File: Library/Formula/makemake.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# makemake formula
```
Comment (DomT4) [inline Library/Formula/makemake.rb]: All of the commented out stuff in this PR is fine to cut.

File: Library/Formula/makemake.rb [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+class Makemake < Formula
+
+  homepage "https://github.com/Julow/makemake"
+
+  url "https://github.com/Julow/makemake.git", :revision => "a11bba0903aad98765b7fdac0a67ed09903ba669"
```
Comment (DomT4) [inline Library/Formula/makemake.rb]: If there's no stable release this is going to need to go in `Homebrew/head-only`.

File: Library/Formula/makemake.rb [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+    mv "makemake.py", "makemake"
+    bin.install "makemake"
+
+  end
+
+  test do
+
+    system "echo", "Makemake is an interative program"
+    system "makemake", "-h"
```
Comment (bfontaine) [inline Library/Formula/makemake.rb]: Could you add a test that does more than `makemake -h` or `makemake --version`? Thanks!
Also, you can remove the `system "echo", "Makemake is an interative program"` line as it doesn’t test `makemake`.

**Comment by bfontaine** [comment]
Unfortunately I don’t think this meets our [formulae requirements](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Acceptable-Formulae.md#niche-or-self-submitted-stuff) as it’s a self-submission and doesn’t have stable releases. I’ll leave this open in case a maintainer has another opinion on this. cc @Homebrew/owners @DomT4

File: Library/Formula/makemake.rb [hunk 10]
```diff
@@ -0,0 +1,15 @@
+class Makemake < Formula
+  homepage "https://github.com/Julow/makemake"
+  url "https://github.com/Julow/makemake.git", :tag => "1.0-release"
```
Comment (bfontaine) [inline Library/Formula/makemake.rb]: Please specify a `:revision` along with the tag, which prevents retagging issues.

File: Library/Formula/makemake.rb [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+  head "https://github.com/Julow/makemake.git"
+  sha256 "47b79f953e39db26d4cdd612572c52e9a06b7899025ef53ed874baa4a7a03b23"
+  version "1.0"
```
Comment (bfontaine) [inline Library/Formula/makemake.rb]: You can remove this line as Homebrew guesses the version from the tag above.

File: Library/Formula/makemake.rb [hunk 12]
```diff
@@ -0,0 +1,15 @@
+class Makemake < Formula
+  homepage "https://github.com/Julow/makemake"
```
Comment (bfontaine) [inline Library/Formula/makemake.rb]: Could you add a short description, using `desc`? The string `makemake: <description>` should be under 80 chars. Thanks!

For example:

```ruby
desc "Makefile generator"
```

File: Library/Formula/makemake.rb [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+  url "https://github.com/Julow/makemake.git", :tag => "1.0-release", :revision => "f3ebda821b4e859cfdd20b579d1074c86330d14f"
+  head "https://github.com/Julow/makemake.git"
+  sha256 "47b79f953e39db26d4cdd612572c52e9a06b7899025ef53ed874baa4a7a03b23"
+  version "1.0"
+  def install
+    mv "makemake.py", "makemake"
+    bin.install "makemake"
```
Comment (bfontaine) [inline Library/Formula/makemake.rb]: You can merge the two lines using `bin.install "makemake.py" => "makemake"`.

File: Library/Formula/makemake.rb [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+  end
+  test do
+    system "makemake", "--test"
```
Comment (bfontaine) [inline Library/Formula/makemake.rb]: Please use `"#{bin}/makemake"` instead of `"makemake"`.

File: Library/Formula/makemake.rb [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+    system "cat", "Makefile"
```
Comment (bfontaine) [inline Library/Formula/makemake.rb]: You can use `assert File.exist? "Makefile"` to ensure a file exists.

**Comment by Julow** [comment]
okok why so strict ?

**Comment by Julow** [comment]
The desc thing does not work
And now it's in a comment because comment are good and minified files are not

File: Library/Formula/makemake.rb [hunk 16]
```diff
@@ -0,0 +1,15 @@
+class Makemake < Formula
+  homepage "https://github.com/Julow/makemake"
+  # desc "Cool Makefile generator (C/C++)"
```
Comment (bfontaine) [inline Library/Formula/makemake.rb]: `desc` is part of the DSL, like `homepage` and `url`. The formula should look like this:

```ruby
class Makemake < Formula
  desc "the description"
  homepage "https://github.com/Julow/makemake"
  url "..."
  # ...
```

**Comment by Julow** [comment]
Error: undefined method `desc' for Formulary::Formulae::Makemake:Class

**Comment by bfontaine** [comment]
You can run `brew audit --strict makemake` locally and fix the warnings without having to wait for the bot or me to check :wink:

**Comment by bfontaine** [comment]
Regarding your `desc` issue, run `brew update` to ensure you have Homebrew’s latest version.

**Comment by Julow** [comment]
Incorrect file permissions: chmod 644 /home/juloo/.brew/Library/Formula/makemake.rb

Cannot find the correct mode in the docs

**Comment by Julow** [comment]
(And it is already 644)

**Comment by DomT4** [comment]
Yeah, I'm a little :-1: on this for now. I agree it could be a very useful project but it's a self-submission and relatively new and niche. I'd be happy to see it maintained in your own tap for a while, and resubmitted in the future if popularity increased.

**Comment by mikemcquaid** [comment]
Passing on this: it's pretty new, self-submitted and not that popular yet. Sorry!

**Pull request closed without merging**